python/covid_graphs/covid_graphs/predictions.py

```diff
@@ -196,5 +204,77 @@ def predictions_for_event(self, prediction_event: PredictionEvent) -> List[Count
     def predictions_for_country(self, country: str) -> List[CountryPrediction]:
         return [p for p in self._prediction_database if p.country == country]
 
+    def select_predictions(
+        self, country: str, last_data_dates: List[datetime.date]
+    ) -> List[CountryPrediction]:
+        return [
+            p
+            for p in self._prediction_database
+            if p.country == country and p.prediction_event.last_data_date in last_data_dates
```

Comment:
I think we should have a dictionary of predictions per country to speed this up.

Also, isn't `in last_data_dates` too slow? Or we don't care? I'd at least convert it to a `set`.

Reply:
I don't think this is the bottleneck (graph generation is). We can optimise in another PR if we want to.

Reply:
The dictionary is a bigger change, but making it into a set is more important and just a one/two line change.

It could be a bottleneck with 28 days selected (see my other suggestion below).

Reply:
I measured it; there's no noticeable effect.